tests/unit/processor/dissecter/test_dissecter.py

```diff
@@ -0,0 +1,473 @@
+# pylint: disable=missing-docstring
+import pytest
+from tests.unit.processor.base import BaseProcessorTestCase
+from logprep.processor.base.exceptions import ProcessingWarning
+
+test_cases = [  # testcase, rule, event, expected
```

Comment:
Maybe add some tests that are trying to break it with special characters like: 
```python
    (
        "my tests",
        {
            "filter": "message",
            "dissecter": {"mapping": {"message": "%{field1} %{field2} %{field3} %{+field4}"}},
        },
        {"message": "This is a + message"},
        {
            "message": "This is \\a + message",
            "field1": "This",
            "field2": "is",
            "field3": "\\a",
            "field4": "+ message",
        },
    ),
```
or
```python
    (
        "my tests",
        {
            "filter": "message",
            "dissecter": {"mapping": {"message": "%{~field1} %{fie ld2} %{-fie}ld3} %{+field4}"}},
        },
        {"message": "&This is\2 a mess}age /1"},
        {
            "message": "&This is\2 a mess}age /1",
            "~field1": "&This",
            "fie ld2": "is\2",
            "-fie}ld3": "a",
            "field4": "mess}age /1",
        },
    ),
```

maybe it is also helpful to have some form of validation for the dissect pattern? At least this rule
```python
        {
            "filter": "message",
            "dissecter": {"mapping": {"message": "%{field1} %{&field2} %{field3} %{+field4}"}},
        },
```
which I think is not a valid pattern results in `AttributeError: 'NoneType' object has no attribute 'split'`.

Reply:
there is a validation for the pattern in the config class of the rule via match_regex. But I will revise it

Reply:
done